initca/initca.go

```diff
@@ -179,6 +227,51 @@ func NewFromSigner(req *csr.CertificateRequest, priv crypto.Signer) (cert, csrPE
 	return
 }
 
+// RenewFromSigner re-creates a root certificate from the CA cert and crypto.Signer.
+// The resulting root certificate will have ca
+// as the template and have the same expiry length.
+func RenewFromSigner(ca *x509.Certificate, priv crypto.Signer) ([]byte, error) {
+	if !ca.IsCA {
+		return nil, errors.New("input certificate is not a CA cert")
+	}
+
+	// it is not within renew window of expiration, don't re-generate
+	if time.Now().AddDate(0, 0, RenewWindowInDays).Before(ca.NotAfter) {
```

Comment:
Trying to only have expiring CA cert be re-generated. I am open to remove it. I guess it is better to let caller decide whether to call Renew.